src/FirmwarePlugin/FirmwarePlugin.cc

```diff
@@ -740,3 +727,35 @@ void FirmwarePlugin::_versionFileDownloadFinished(QString& remoteFile, QString&
         qgcApp()->showMessage(message.arg(vehicle->firmwareVersionTypeString(), currentVersion, version));
     }
 }
+
+int FirmwarePlugin::versionCompare(Vehicle* vehicle, int major, int minor, int patch)
+{
+    int currMajor = vehicle->firmwareMajorVersion();
+    int currMinor = vehicle->firmwareMinorVersion();
+    int currPatch = vehicle->firmwarePatchVersion();
+
+    if (currMajor == major && currMinor == minor && currPatch == patch) {
+        return 0;
+    }
+
+    if (currMajor > major
+       || (currMajor == major && currMinor > minor)
+       || (currMajor == major && currMinor == minor && currPatch > patch))
+    {
+        return 1;
+    }
```

Comment:
Is it more simple to compare:
```
currMajor >= major
currMinor >= minor
currPatch > patch
```

Reply:
could you elaborate?
Doesn't look like it would work comparing 4.0.0 with 3.0.0

Reply:
I need to stop reviewing stuff at night.